### PR TITLE
fix: Update to use JSON tool definitions

### LINKED ISSUE #1263: Update rtl-demo to use JSON toolbox definitions instead of xml
**Category**

 - Examples

**Component**

rtl-demo

**Is your feature request related to a problem? Please describe.**

Many of the examples and plugins use XML toolboxes instead of the new JSON API. We should have our example code use the new APIs to best model what other developers can do with Blockly.

**Describe the solution you'd like**

Update the code in blockly-samples to use the JSON toolbox API.

**Additional context**

To work on this issue, first read the [background information](https://developers.google.com/blockly/guides/contribute/samples) about how to contribute to samples and verify your changes.

1. Learn about [defining a toolbox](https://developers.google.com/blockly/guides/configure/web/toolbox) and read about the XML format vs the JSON format.
2. In blockly-samples, navigate to the rtl-demo example's directory. Some examples can be started with `npm run start` and some can be opened directly by opening the `index.html` file in a browser. The README may give specific instructions, or you can look in the `package.json` to see what scripts are defined. Note you may need to run `npm install` in the example directory. Either way, open the example's demo page to observe the toolbox and its contents.
3. Write a new copy of the toolbox definition in the JSON format instead of the XML format. Where exactly it should be declared will depend on the example. It will most likely make sense to declare the toolbox definition close to the location where it is used, i.e. where `Blockly.inject` is called.
4. Refresh the demo page and make sure the toolbox looks the same as it did before. You can compare against the version in master on the [samples site](https://google.github.io/blockly-samples/) to make sure the behavior hasn't changed. Note that not all demos are published on the samples site so it may not be there.
5. Follow the rest of the instructions in the Contributing docs to verify your change and open a PR.

### PR DESCRIPTION
**Pull request by karenjohn35**
## The basics

- [x] I [validated my changes](https://developers.google.com/blockly/guides/contribute/samples#making_and_verifying_a_change)

### Resolves

Fixes #1263

### CODE REVIEW

File: examples/rtl-demo/index.html [hunk 1]
```diff
@@ -23,15 +23,364 @@
   </style>
   <script>
     function init() {
-      var demoWorkspace = Blockly.inject('blocklyDiv',
-          {media: './node_modules/blockly/media/',
-           rtl: true,
-           toolbox: document.getElementById('toolbox')});
+      var toolbox = {
```
Comment (BeksOmega) [inline examples/rtl-demo/index.html]:
```suggestion
      const toolbox = {
```

File: examples/rtl-demo/index.html [hunk 2]
```diff
@@ -23,15 +23,364 @@
...
+          {
+            "kind": "category",
+            "name": "إجراءات",
+            "custom": "PROCEDURE",
+            "colour": "%{BKY_PROCEDURES_HUE}",
+          },
+        ]
+      }
+  
+      var workspace = Blockly.inject('blocklyDiv',
```
Comment (BeksOmega) [inline examples/rtl-demo/index.html]:
```suggestion
      const workspace = Blockly.inject('blocklyDiv', 
```

**Review by BeksOmega — CHANGES_REQUESTED** [review]
This looks great! Just two little fixes then should be good to go =)

Thank you for your hard work on this!

File: examples/rtl-demo/index.html
Comment (debaraj-barua) [inline examples/rtl-demo/index.html:1]: @karenjohn35 Perhaps L390-L515 can be removed?

https://github.com/google/blockly-samples/blob/772b45371cd1fa02d9feeade8be28ddce318dd32/examples/rtl-demo/index.html#L390-L515

Comment (BeksOmega) [reply examples/rtl-demo/index.html:1]: That would be great as well =) Thanks for the catch @debaraj-barua !

**Comment by BeksOmega** [comment]
Heya @karenjohn35  Just wanted to check that you're still interested in working on this! Otherwise I'm going to close it on Friday 2023/10/06 so we don't have stale PRs =) Thank you again for your work on this!

File: examples/rtl-demo/index.html
Comment (karenjohn35) [reply examples/rtl-demo/index.html:1]: How can I accept these changes?

**Comment by karenjohn35** [comment]
Hi @BeksOmega! I think you can resolve this pull request after I implement the changes @debaraj-barua proposed. I am not entirely sure how I can do that. Would I have to create another pull request?

**Comment by BeksOmega** [comment]
You would need to:

1) Check out your OSD branch. `git switch osd`
2) Delete the lines debaraj suggested
3) Commit the changes. `git commit -am "fix: delete xml"`
4) Push the changes to your remote. `git push origin osd`

Then I can get this merged!

File: examples/rtl-demo/index.html
Comment (BeksOmega) [reply examples/rtl-demo/index.html:1]: The UI here is a bit confusing, this isn't a suggested edit, just a link to the relevant lines of code!

**Comment by karenjohn35** [comment]
Thanks, I just made that edit!

**Review by BeksOmega — APPROVED** [review]

**Comment by BeksOmega** [comment]
Thank you so much for the work on this @karenjohn35 This looks great :D

**Pull request merged**